app/subscriber/src/features/my-reports/view/ReportEdit.tsx

```diff
@@ -75,12 +75,13 @@ export const ReportEdit: React.FC = () => {
   }, []);
 
   // Case 2: Some report data loaded in store, check if current report is present
-  // if it is, set it to that report, if not, try to fetch it
+  // AND has content, if so, set it to that report, if not, fetch report w/ content
   React.useEffect(() => {
     const reportId = parseInt(id ?? '0');
     if (!!reportId && myReports?.length) {
       const existingReport = myReports.find((r) => r.id === reportId);
-      if (existingReport) {
+      const hasFetchedContent = existingReport?.instances.some((r) => r.content?.length);
```

Comment:
This is the best I can do given the current data structure. I am assuming that if there's _any_ content present for any of the instances, that means we have fetched content for this report before, so there's no need to refetch

Here's the rough breakdown of what data is present & when:

- fetchMyReports() -> returns reports with no instance data, so `instances: []`
- getReport(id) -> returns report with instances data but NO content, `instances: [{ ...instanceData, content: [] }]`
- getReport(id, true) -> returns report + instance + **content**
- generateReport(id) -> reports instance + **content**

As long as one of `getReport(id, true)` or `generateReport(id)` have been called by this point. All content data needed should be present.

Reply:
I think it's a reasonable approach.